elements/howto-accordion/demo.html

```diff
@@ -0,0 +1,44 @@
+<!doctype html>
+<style>
+  howto-accordion {
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: column;
+    align-items: stretch;
+  }
+  /* Headings need a background color so they are not transparent. Otherwise
+  the content will shine through during the animation. */
+  howto-accordion-heading {
+    display: flex;
+  }
+  howto-accordion-heading > button {
+    flex-grow: 1;
+    background-color: white;
+    border: 1px solid black;
+  }
+  howto-accordion-heading[expanded] > button {
+    background-color: bisque;
+  }
+  howto-accordion-heading + * {
+    padding: 20px;
+    background-color: lightgray;
+  }
+  /* The accordion gets the `animating` class assigned if its children are about
+  to be animated. The implementation relies on the `transitionend` event, so
+  this class _must_ define a transition on the `transform` property. */
+  howto-accordion > *.animating {
+    transition: transform 0.3s ease-in-out;
+  }
+  howto-accordion-heading + .collapsed {
+    display: none;
+  }
+</style>
+
+<howto-accordion>
+  <howto-accordion-heading><button>Tab 1</button></howto-accordion-heading>
```

Comment:
It would abstract a lot of weirdness. I am a little worried about the styling without JAvaScript, but I’ll give that a go as it will also address loads of your other comments.